### ISSUE TITLE
Detect missing ellipses in protocols

### ISSUE THREAD

**Issue by rpmrmartin** [microsoft/pyright#4382]
**Is your feature request related to a problem? Please describe.**
If I forget the ellipsis after a method definition in a protocol, Pyright neither detects missing implementations in derived classes nor detects that anything is wrong with the protocol.

**Describe the solution you'd like**
Either Pyright should detect missing implementations in derived classes even if a method definition in a protocol as missing the ellipsis or it should detect that an ellipsis is missing from the protocol.

**Additional context**
Here's a simple test case. I would like Pyright to detect that something is wrong with either `ICombobulate` or `Combobulator`, but Pyright detects no errors in this code.
```python
from typing import Protocol

class ICombobulate(Protocol):
	def combobulate(self) -> None:
		'''Combobulate'''

class Combobulator(ICombobulate):
	pass
```

**Comment by erictraut** [comment]
The current behavior is "as designed", but I think it's worth some additional discussion. The real question is what constitutes an "unimplemented placeholder". These are useful in abstract methods, protocols, and other cases where the function or method is expected never to be executed.

We need to consider four cases. A function body whose contents contain:
1. Only an ellipsis
2. A docstring plus an ellipsis
3. Only a docstring
4. Some other statement that is not an ellipsis or a docstring

I think we would all agree that 1 and 2 should be treated as placeholder implementations and 4 should not. Case 3 is the tricky one.

If we say that Case 3 should be treated as a placeholder implementation, then pyright will not enforce return type consistency and it will complain if an abstract or protocol method is not implemented in a subclass. If we say that Case 3 should not be treated as a placeholder implementation (as it is now), then pyright will enforce return type consistency but will not complain about abstract or protocol method overrides.

Given the current interpretation, the following will generate a return type consistency error, which I think is correct.

```python
def foo() -> int:
    """ I forgot to add a body """
```

In your case, you left off the `...`, so it was considered an implementation. The return type was annotated as `None`, so pyright detected no return type compatibility issue.

Between these two options, I think I slightly prefer the current behavior. Maybe a compromise is to emit a separate error _specifically for abstract methods and methods in a protocol_ whose body contains only a docstring but no ellipsis.

**Comment by rpmrmartin** [comment]
I did not realize that protocols could provide default implementations, but I can see now that the definition of `combobulate` in my example is valid, so Pyright may not want to complain about it. Perhaps this check is better implemented by some other tool, like a linter.

**Comment by erictraut** [comment]
Yeah, after thinking about it a bit more, I think it's probably best to leave the behavior as is. I'm open to revisiting this in the future if we get additional feedback, but I'm going to close this for now.
